src/lib/lldpctl.hpp

```diff
@@ -235,6 +235,7 @@ class LldpAtom {
 		{
 			list.emplace_back(atom, true, conn_);
```

Comment:
Here we need to add a reference to `this` to extend its lifetime as long as the atoms in the list are alive:
```suggestion
			list.emplace_back(atom, true, conn_, std::make_unique<LldpAtom>(*this));
```

Reply:
Yes, you are right, if the children cannot live without its parent atom. Updated as you suggested.